sphinx_gallery/gen_rst.py

```diff
@@ -647,6 +641,11 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
                 example_rst += bcontent + '\n'
                 example_nb.add_markdown_cell(text2string(bcontent))
 
+    # Writes md5 checksum if example has build correctly
+    if not src_file in gallery_conf['failing_examples']:
+        with open(example_file + '.md5', 'w') as file_checksum:
+            file_checksum.write(get_md5sum(example_file))
```

Comment:
You are right it's not that easy to do actually. I was hoping you could do this only where no error happens, but it's not that simple for notebook-like examples since you need to keep iterating over the remaining blocks even if an error occured ...

I wasn't a big fan of the gallery_conf being used as a way to keep track of the failing examples for the end of build summary, but I am even less of a fan now that it is being used to dictate the logic in quite a few places ... After this PR maybe we should spend some time trying to clean this up.